packages/components/src/navigator/navigator-back-link/component.tsx

```diff
@@ -0,0 +1,65 @@
+/**
+ * External dependencies
+ */
+// eslint-disable-next-line no-restricted-imports
+import type { Ref } from 'react';
+
+/**
+ * WordPress dependencies
+ */
+import { useCallback } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import {
+	contextConnect,
+	useContextSystem,
+	WordPressComponentProps,
+} from '../../ui/context';
+import { View } from '../../view';
+import useNavigator from '../use-navigator';
+import type { NavigatorBackLinkProps } from '../types';
+
+function NavigatorBackLink(
+	props: WordPressComponentProps< NavigatorBackLinkProps, 'button' >,
+	forwardedRef: Ref< any >
+) {
+	const {
+		children,
+		onClick,
+		as = 'button',
```

Comment:
Why is this a "button" by default and not a "Button"? Feels like an opportunity for more composition to me.

Reply:
Done in https://github.com/WordPress/gutenberg/pull/37223/commits/d219320891d6554e5a09b422a3c171cea5158412